### PR TITLE
If FW landing speed -1, set to halfway between min and trim aspd; FW only autolands

### PR DESCRIPTION
**Pull request by Hwurzburg**
alternate to #23601 that affects only FW autolanding approach speed since a group is working on VTOL approach/speed

### CODE REVIEW

**Review by IamPete1 — CHANGES_REQUESTED** [review]
If `land_airspeed` is not set we will use this new value for the approach stage and then speed up to cruise airspeed again in the preflair and final stages?

I think this new value should be used in those stages too.

**Comment by Hwurzburg** [comment]
and it save some flash too...thanks

**Comment by IamPete1** [comment]
There is a trap in that switch case, there is also a `SlopeStage::NORMAL` we should still use cruise speed for that I think. You if you added that to the switch you could remove the default case.

**Comment by Hwurzburg** [comment]
That case doesn't exist, This function only gets called in those 3 cases

Sent from Yahoo Mail on Android 
 
  On Sun, Jun 25, 2023 at 2:54 PM, Peter ***@***.***> wrote:   


There is a trap in that switch case, there is also a SlopeStage::NORMAL we should still use cruise speed for that I think. You if you added that to the switch you could remove the default case.

File: libraries/AP_Landing/AP_Landing_Slope.cpp [hunk 1]
```diff
@@ -346,26 +346,24 @@ int32_t AP_Landing::type_slope_get_target_airspeed_cm(void)
 
     const float land_airspeed = tecs_Controller->get_land_airspeed();
     int32_t target_airspeed_cm = aparm.airspeed_cruise_cm;
```
Comment (IamPete1) [inline libraries/AP_Landing/AP_Landing_Slope.cpp:348]:
```suggestion
    int32_t target_airspeed_cm;
```

File: libraries/AP_Landing/AP_Landing_Slope.cpp [hunk 2]
```diff
@@ -346,26 +346,24 @@ int32_t AP_Landing::type_slope_get_target_airspeed_cm(void)
...
-
+    if (land_airspeed >= 0) {
+        target_airspeed_cm = land_airspeed * 100;
+    } else {
+        target_airspeed_cm = 0.5 * (aparm.airspeed_cruise_cm * 0.01 + aparm.airspeed_min);
```
Comment (IamPete1) [inline libraries/AP_Landing/AP_Landing_Slope.cpp:352]: Sorry, just spotted this. Airspeed min is in m/s.

```suggestion
        target_airspeed_cm = 0.5 * (aparm.airspeed_cruise_cm + aparm.airspeed_min * 100.0);
```

**Review by IamPete1 — DISMISSED** [review]

File: libraries/AP_Landing/AP_Landing_Slope.cpp (hunk 1 above)
Comment (Hwurzburg) [reply libraries/AP_Landing/AP_Landing_Slope.cpp:348]: at least its not mine...

Comment (Hwurzburg) [reply libraries/AP_Landing/AP_Landing_Slope.cpp:348]: although I should have recognized it

**Review by tridge — APPROVED** [review]

**Pull request merged**